test/test_memory.cpp

```diff
@@ -75,4 +75,144 @@ TEST_F(MemoryTest, LoadCache) {
       &fusion, actual_ts, {input}, {expected_output}, __LINE__, __FILE__);
 }
 
+class TMATest : public NVFuserTest {
+  void SetUp() override {
+    // requires Hopper or newer
+    if (!deviceMajorMinorCheck(9)) {
+      GTEST_SKIP() << "skipping tests on pre-Hopper GPUs";
+    }
+    NVFuserTest::SetUp();
+  }
+};
+
+TEST_F(TMATest, StoreCompleteTensor1D) {
```

Comment:
There is an unused variable in the generated code, e.g. `a1`
```
__global__ void kernel1(Tensor<float, 1, 1> T0, const __grid_constant__ TensorMap var0, Tensor<float, 1, 1> T2) {
  alignas(16) extern __shared__ char array[];
  const unsigned smem_offset = 0;
  Array<nvfuser_index_t, 1, 1> a1;
  float* T1 = reinterpret_cast<float*>(array + smem_offset + 0);
  #pragma unroll 1
  for(nvfuser_index_t i2 = 0; i2 < T0.logical_size[0]; ++i2) {
    T1[i2]
       = T0[i2];
  }
  Hopper::cpAsyncBulkTensorTileS2G((Hopper::CpAsyncBulkTensorTileIndex<1>{ (&var0), (Array<nvfuser_index_t, 1, 1>{0}) }), toSmem(T1));
}
```

Reply:
Yeah, this is because the index hoisting in this PR is just a hack: https://github.com/NVIDIA/Fuser/pull/748/files#r1315401800
It should disappear when index hoisting is better implemented.